osf_tests/factories.py

```diff
@@ -1142,12 +1141,15 @@ def _create(cls, *args, **kwargs):
 
         registration.registered_schema.clear()
         registration.registered_schema.add(schema)
-
-        if SchemaResponse.objects.filter(object_id=registration.id, content_type_id=ContentType.objects.get_for_model(registration)).exists():
+        registration.save()
+        content_type = ContentType.objects.get_for_model(registration)
+        if SchemaResponse.objects.filter(object_id=registration.id, content_type_id=content_type).exists():
```

Comment:
This can just be
```
if registration.schema_responses.exists():
  previous_response = registration.schema_responses.first()
  . . .
```
alternatively, mix up the the logic:
```
previous_response =registration.schema_responses.first()
if previous_response:
 . . .
if not previous_response:
 . . .
```

Using `.get()` assumes that *only* the initial resposnes exist, which would not be the case if using a SchemaResposneFactory for the same registration multiple times